nixos/modules/services/web-servers/keter/default.nix

```diff
@@ -0,0 +1,163 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.keter;
+in
+{
+  meta = {
+    maintainers = with lib.maintainers; [ jappie ];
+  };
+
+  options.services.keter = {
+    enable = lib.mkEnableOption ''keter, a web app deployment manager.
+        Note that this module only support loading of webapps:
+        Keep an old app running and swap the ports when the new one is booted.
+    '';
+
+    keterRoot = lib.mkOption {
+      type = lib.types.str;
+      default = "/var/lib/keter";
+      description = ''Mutable state folder for keter'';
+    };
+
+    keterPackage = lib.mkOption {
+      type = lib.types.package;
+      default = null;
+      description = ''The keter package to be used, usually <varname>pkgs.haskellPackages.keter</varname>'';
```

Comment:
No crossover from `defaultText` please.

Reply:
I set the defaultText instead of varname, I also set the default to `pkgs.haskellPackages.keter` I'm not sure why I thought I had to set this to null before, it just works now.